spring-shell2-core/pom.xml

```diff
@@ -0,0 +1,16 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<artifactId>spring-shell2-core</artifactId>
+	<packaging>jar</packaging>
+
+	<parent>
+		<groupId>org.springframework.shell</groupId>
```

Comment:
Coming back to your core vs api proposal, I thought about it a bit more and may now see some value to it.
There are two user profiles here, I think:
- people who want to write new adapters
- and the more, regular, end user profile who just wants to write commands (certainly using the "standard" set of annotations). 
Having 2 artifacts for that may indeed be a better approach.

Don't worry though, this can be done in a second step.
I'm in favor of merging this PR asap, obviously, given how it hinders others